subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -536,49 +526,25 @@ public ProjectStateInternal getState() {
         return state;
     }
 
-    @Override
-    public FileResolver getFileResolver() {
-        if (fileResolver == null) {
-            fileResolver = services.get(FileResolver.class);
-        }
-        return fileResolver;
-    }
+    @Override @Inject
+    public abstract FileResolver getFileResolver();
 
-    @Override
-    public TaskDependencyFactory getTaskDependencyFactory() {
-        if (taskDependencyFactory == null) {
-            taskDependencyFactory = services.get(TaskDependencyFactory.class);
-        }
-        return taskDependencyFactory;
-    }
-
-    public void setFileResolver(FileResolver fileResolver) {
-        this.fileResolver = fileResolver;
-    }
+    @Override @Inject
+    public abstract TaskDependencyFactory getTaskDependencyFactory();
 
     public void setAnt(AntBuilder ant) {
         this.ant = ant;
     }
 
-    @Override
-    public ArtifactHandler getArtifacts() {
-        if (artifactHandler == null) {
-            artifactHandler = services.get(ArtifactHandler.class);
-        }
-        return artifactHandler;
-    }
+    @Override @Inject
+    public abstract ArtifactHandler getArtifacts();
 
     @Inject
     @Override
     public abstract RepositoryHandler getRepositories();
 
-    @Override
-    public RoleBasedConfigurationContainerInternal getConfigurations() {
-        if (configurationContainer == null) {
-            configurationContainer = services.get(RoleBasedConfigurationContainerInternal.class);
-        }
-        return configurationContainer;
-    }
+    @Override @Inject
```

Comment:
💅 This is the third variation in the `DefaultProject` class on how to add the `@Inject` annotation to the method.
I like the extra line saved, but maybe keeping it uniform (put them on separate lines) would be still better.

Reply:
Done